addons/website_payment/models/payment_acquirer.py

```diff
@@ -16,5 +17,6 @@ class PaymentAcquirer(models.Model):
     def get_base_url(self):
         # Give priority to url_root to handle multi-website cases
         if request and request.httprequest.url_root:
-            return request.httprequest.url_root
+            # If necessary, encode non-ascii domains to punycode.
```

Comment:
Try to explain why we need to use `iri_to_uri`.
```suggestion
            # Some domain names can use non-Latin script or alphabet or the Latin alphabet-based characters with diacritics or ligatures. They are stored as ASCII strings using Punycode transcription in the DNS system and need to be converted to send to external APIs.
```

Reply:
@chevalierv updated :+1: Thanks for the check!